tests/test_inspection.py

```diff
@@ -206,6 +208,73 @@ def test_invalid_npk(self):
                     inspection.npk, f"Expected None for npk with input {npk}"
                 )
 
+class TestRegistrationNumber(unittest.TestCase):
+    def setUp(self):
+        self.valid_registration_number_data = [
+            "1234567A",
+            "1234567B",
+            "1234567C",
+            "1234567D",
+            "1234567E",
+            "1234567F",
+            "1234567G",
+            "1234567H",
+            "1234567I",
+            "1234567J",
+            "1234567K",
+            "1234567L",
+            "1234567M",
+            "1234567N",
+            "1234567O",
+            "1234567P",
+            "1234567Q",
+            "1234567R",
+            "1234567S",
+            "1234567T",
+            "1234567U",
+            "1234567V",
+            "1234567W",
+            "1234567X",
+            "1234567Y",
+            "1234567Z",
+        ]
+
+        self.invalid_registration_number_data = [
+            "1234567",
+            "1234567AA",
+            "1234567A1",
+            "1234567A ",
+            "1234567 A",
+            "1234567A-",
+            "1234567A.",
+            "1234567A,",
+        ]
+
+        self.invalid_registration_type_data = [
+            "FERTILIZER",
+            "INGREDIENT",
+            "PRODUCT",
+            "COMPONENT",
+        ]
+
+    def test_valid_registration_number(self):
+        for registration_number in self.valid_registration_number_data:
+            with self.subTest(registration_number=registration_number):
+                registration = RegistrationNumber(identifier=registration_number)
+                self.assertEqual(registration.identifier, registration_number)
+            
+    def test_invalid_registration_number(self):
+        for registration_number in self.invalid_registration_number_data:
+            with self.subTest(registration_number=registration_number):
+                with self.assertRaises(ValueError):
+                    RegistrationNumber(identifier=registration_number, type=RegistrationNumberType.FERTILIZER)
+
+    def test_invalid_registration_type(self):
+        for registration_type in self.invalid_registration_type_data:
+            with self.subTest(registration_type=registration_type):
+                with self.assertRaises(ValueError):
+                    RegistrationNumber(identifier="1234567A", type=registration_type)
+    
```

Comment:
Did you also test if the pipeline is able to extract different type of registration numbers?

Reply:
All registration numbers have the same format.

Reply:
I know, but the model should figure out what type of registration number it is, is it fertilizer or ingridient.